### PR TITLE
[WIP] Clean up and document build and runtime configuration

### PR DESCRIPTION
**Pull request by jsenko**
I went over build and runtime configuration options (incl. JPA), and made sure users know how to run the app.
Please take a look at README.md and check it makes sense and the information is valid.
@alesj Could you please add the kafka part? Thanks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,86 @@
...
+To see additional options, visit:
+ - [Data Source options](https://quarkus.io/guides/datasource-guide#configuration-reference) 
+ - [Hibernate options](https://quarkus.io/guides/hibernate-orm-guide#properties-to-refine-your-hibernate-orm-configuration)
+    
+### Kafka
+
+*TODO*
+
+### Docker container
+The same options are available for the docker containers, but only in the form of environment variables (The command line parameters are for the `java` executable and at the moment it's not possible to pass them into the image).
```
Comment (jsenko) [inline README.md]: s/image/container

File: README.md [hunk 2]
```diff
@@ -1,4 +1,86 @@
...
+ - `-Ddocker` *(experimental)* builds docker images. Make sure that you have the docker service enabled and running.
+   If you get an error, try `sudo chmod a+rw /var/run/docker.sock`.
+
+## Runtime Configuration
+
+The following parameters are available for executable files:
+
+### JPA
+ - In the *dev* mode, the application expects a H2 server running at `jdbc:h2:tcp://localhost:9123/mem:registry`.
+ - In the *prod* mode, you have to provide connection configuration to a PostgreSQL server as follows:
```
Comment (jsenko) [inline README.md]: s/to a PostgreSQL server/for a PostgreSQL server

File: pom.xml [hunk 3]
```diff
@@ -107,7 +107,7 @@
 		<!-- Confluent API / client -->
 		<confluent.version>5.3.0</confluent.version>
 
-		<!-- Build config -->
+		<!-- Default Quarkus build profile -->
```
Comment (jsenko) [inline pom.xml]: s/build/configuration

File: pom.xml [hunk 4]
```diff
@@ -381,6 +381,15 @@
 				<module>storage/kafka</module>
 			</modules>
 		</profile>
+		<profile>
+			<id>prod</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
```
Comment (jsenko) [inline pom.xml]: true -> false

File: app/src/main/resources/application.properties [hunk 5]
```diff
@@ -35,3 +32,38 @@ quarkus.index-dependency.jaxrs.artifact-id=jboss-jaxrs-api_2.1_spec
...
+# === Prod profile - see README
+
+# Log
+%prod.quarkus.log.level=INFO
+%prod.quarkus.log.console.enable=true
+
+# JPA
+%prod.quarkus.datasource.driver=org.postgresql.Driver
+%prod.quarkus.datasource.url=${QUARKUS_DATASOURCE_URL}
+%prod.quarkus.datasource.username=${QUARKUS_DATASOURCE_USERNAME:}
```
Comment (EricWittmann) [inline app/src/main/resources/application.properties]: Why the colon after USERNAME but not after URL?  optional?

Comment (jsenko) [reply app/src/main/resources/application.properties]: The data source URL is required to connect to the DB

Comment (jsenko) [reply app/src/main/resources/application.properties]: Others are optional, exactly

**Pull request merged**